src/Core/BufferLine.re

```diff
@@ -33,24 +37,130 @@ type t = {
   indentation: IndentationSettings.t,
   // [raw] is the raw string (byte array)
   raw: string,
+  // [font] is the main font to render the line with
+  font: Font.t,
+  // [skiaTypeface] is the Skia typeface associated with the font
+  skiaTypeface: Skia.Typeface.t,
+  // [glyphStrings] is a list of typefaces and strings from shaping
+  mutable glyphStrings: list((Skia.Typeface.t, string)),
   // [characters] is a cache of discovered characters we've found in the string so far
   mutable characters: array(option(characterCacheInfo)),
   // [byteIndexMap] is a cache of byte -> index
   mutable byteIndexMap: array(option(int)),
-  // nextByte is the nextByte to work from, or -1 if complete
+  // [nextByte] is the nextByte to work from, or -1 if complete
   mutable nextByte: int,
-  // nextIndex is the nextIndex to work from
+  // [nextIndex] is the nextIndex to work from
   mutable nextIndex: int,
-  // nextPosition is the graphical position (based on character width)
-  mutable nextPosition: int,
+  // [nextGlyphStringIndex] is the next nth glyph string in shapes
+  mutable nextGlyphStringByte: int,
+  // [nextCharacterPosition] is the graphical position (based on character width)
+  mutable nextCharacterPosition: int,
```

Comment:
I'm wondering why we need `nextCharacterPosition` with the pixel position?

Reply:
I got rid of all the character measurements after our discussion 👍